lib/Basic/FileSystem.cpp

```diff
@@ -62,6 +62,19 @@ class LocalFileSystem : public FileSystem {
     }
     return std::unique_ptr<llvm::MemoryBuffer>(result->release());
   }
+
+  virtual bool remove(const std::string& path) override {
```

Comment:
I *think* that for state file removal purposes, we need this to be recursive. There is an implementation of that hiding in the BuildSystemTests I think, which Mirza ported from a previous LLVM revision. We should probably elevate that out to be a real utility function (probably deserves a separate PR), then use that.

I'm ok landing this independently and then adding support for removing subtrees, though.